File: scala-demo/src/main/scala/com/gaodashang/demo/MySQLUpperCaseStrategy.scala
package com.gaodashang.demo

import org.hibernate.cfg.ImprovedNamingStrategy

/**
 * comments.
 * @author eva
 */
class MySQLUpperCaseStrategy extends ImprovedNamingStrategy{
  override def columnName(columnName: String): String = {
    val columnLowerCaseName:String = super.columnName(columnName)
    val columnUpperCaseName:String = columnLowerCaseName.toUpperCase
    columnUpperCaseName
  }

  override def tableName(tableName: String): String = {
    super.tableName(tableName).toUpperCase
  }
}

File: hello-world-a/src/main/java/com/gaodashang/demo/MySQLUpperCaseStrategy.java
package com.gaodashang.demo;

import org.hibernate.cfg.ImprovedNamingStrategy;

/**
 * comments.
 *
 * @author eva
 */
public class MySQLUpperCaseStrategy extends ImprovedNamingStrategy {

    public String classToTableName(String className) {
        return super.classToTableName(className).toUpperCase();
    }

    public String propertyToColumnName(String propertyName) {
        return super.propertyToColumnName(propertyName).toUpperCase();
    }

    public String tableName(String tableName) {
        return super.tableName(tableName).toUpperCase();
    }

    public String columnName(String columnName) {
        return super.columnName(columnName).toUpperCase();
    }

}
